### PR TITLE
Minor dev-experience fixes for making spec changes.

### PR DESCRIPTION
**Pull request by jsuereth**
Make it easier to run all the consistency checks on the command line, or apply autofixes.

Note: I don't comment on any of the tools behaviors/reasoning as I didn't look that far in.  Mostly I just wanted for my own workflow:

```
make fix && make check && git push <remote> <my-cool-idea>
```

## Changes

- Add a `check` target to Makefile that will run all the existing checks in order of "how long they take on jsuereth's machine"
- Add a `fix` target to Makefile that will run anything that can modify local code and may be needed for PRs
- Added an `install-tools` to Makefile that tries to run all the tool installation pieces.  Note: This is likely the least useful for now.
- Added a couple TODOs in the Makefile if anyone has ideas/thoughts.

### CODE REVIEW

**Review by iNikem — APPROVED** [review]
Thanks!

File: Makefile [hunk 1]
```diff
@@ -23,6 +24,7 @@ misspell-correction:
 
 .PHONY: install-markdown-link-check
 install-markdown-link-check:
+	# TODO: Check for existence before installing
```
Comment (tigrannajaryan) [inline Makefile:27]: I think npm install is noop if the package is already installed so the todo may not be needed (I may be wrong, JS gurus can confirm).

File: Makefile [hunk 2]
```diff
@@ -31,13 +33,30 @@ markdown-link-check:
...
 .PHONY: install-markdownlint
 install-markdownlint:
+    # TODO: Check for existence before installing
 	npm install -g markdownlint-cli
 
 .PHONY: markdownlint
 markdownlint:
-	@echo $(ALL_DOCS)
+    # Limit output for clearer error messages
+	# @echo $(ALL_DOCS)
```
Comment (tigrannajaryan) [inline Makefile:42]: I think it is safe to delete this instead of commenting out.

File: Makefile [hunk 3]
```diff
@@ -31,13 +33,30 @@ markdown-link-check:
...
+
+# Attempt to fix issues / regenerate tables.
+.PHONY: fix
+fix: table-generation misspell-correction
+	@echo "All autofixes complete"
+
+# Attempt to install all the tools
+.PHONY: install-tools
+install-tools: install-misspell install-markdownlint install-markdown-link-check
+	@echo "All tools installed"
```
Comment (tigrannajaryan) [inline Makefile:62]: Nit: add an EOL so that github's viewer does not scare us with: 
image

**Review by tigrannajaryan — APPROVED** [review]

**Review by carlosalberto — APPROVED** [review]

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -39,6 +39,41 @@ sparing use of imperatives:
...
+### Consistency Checks
+
+The Specification has a number of tools it uses to check things like style,
+spelling and link validity. You can run all of these locally via:
+
+```bash
+make check
+```
+
+Note: This can take a long time as it checks all links. You should use this
```
Comment (yurishkuro) [inline CONTRIBUTING.md:51]: not part of this PR, but it would be awesome to just build a single Docker image with the tools instead of installing each one individually (and not in a particular clean manner, like `npm -g`)

File: Makefile (hunk 1 above)
Comment (jsuereth) [reply Makefile:27]: On my system it requires sudo.  It may be a noop, but it fails asking for permissions to the folder EVEN if the tool is installed (likely due to upgrade checks)

File: Makefile (hunk 2 above)
Comment (jsuereth) [reply Makefile:42]: done

File: CONTRIBUTING.md (hunk 4 above)
Comment (jsuereth) [reply CONTRIBUTING.md:51]: +100.

File: Makefile (hunk 1 above)
Comment (aabmass) [reply Makefile:27]: You can `npx` to run commands from npm packages without root

```
node@f351847fb5cc:/$ npx markdownlint-cli --help
npx: installed 35 in 2.452s
Usage: markdownlint [options] <files|directories|globs>
...

node@f351847fb5cc:/$ npx markdown-link-check --help
npx: installed 65 in 3.8s
Usage: markdown-link-check [options] [filenameOrUrl]
```

**Review by bogdandrutu — APPROVED** [review]

**Comment by bogdandrutu** [comment]
Merging this and will followup with TODO and docker suggestion
